x-pack/plugins/apm/public/components/routing/templates/apm_main_template.tsx

```diff
@@ -29,14 +32,34 @@ export function ApmMainTemplate({
   pageTitle?: React.ReactNode;
   pageHeader?: EuiPageHeaderProps;
   children: React.ReactNode;
-} & EuiPageTemplateProps) {
+} & KibanaPageTemplateProps) {
+  const { http } = useKibana().services;
+  const basePath = http?.basePath.get();
   const { services } = useKibana<ApmPluginStartDeps>();
 
   const ObservabilityPageTemplate =
     services.observability.navigation.PageTemplate;
 
+  // TODO: NEEDS A DATA CHECK
+  const hasData = true;
```

Comment:
The ES query for doing the check for data is here:
https://github.com/elastic/kibana/blob/c6dc6e207a8ed8d30d74f84b9a554b1af5c27d2c/x-pack/plugins/apm/server/lib/services/get_services/has_historical_agent_data.ts#L12-L32

I suggest a new api endpoint is added like `/apm/api/has_data`. Routes can be added to https://github.com/elastic/kibana/tree/c6dc6e207a8ed8d30d74f84b9a554b1af5c27d2c/x-pack/plugins/apm/server/routes

To fetch data use `useFetcher`:
```ts
  const { data, status, error } = useFetcher(
    (callApmApi) => callApmApi({ endpoint: 'GET /api/apm/has_data' }),
    []
  );
```

Reply:
Sorry, that seems to be beyond my technical means to convert to usable code. Would you mind creating a PR against this branch? Or write out explicitly what needs to be done to achieve this?

Reply:
Totally makes sense. Is this still scheduled for 7.15? FF is today and it would be easier for us to plan it into the 7.16 schedule.

Reply:
Thanks, yeah it looks like this won't be getting in for 7.15. I'd appreciate all the help to get this one over the line after FF.